tracing/tracing.hh

```diff
@@ -148,19 +148,35 @@ public:
     uint32_t slow_query_threshold_us; // in microseconds
     uint32_t slow_query_ttl_sec; // in seconds
     span_id parent_id;
+    bool classic_tracing;
+    bool opentelemetry_tracing;
 
 public:
-    trace_info(utils::UUID sid, trace_type t, bool w_o_c, trace_state_props_set s_p, uint32_t slow_query_threshold, uint32_t slow_query_ttl, span_id p_id)
+    trace_info(utils::UUID sid, trace_type t, bool w_o_c, trace_state_props_set s_p, uint32_t slow_query_threshold, uint32_t slow_query_ttl, span_id p_id, bool otel_tracing = false)
         : session_id(std::move(sid))
         , type(t)
         , write_on_close(w_o_c)
         , state_props(s_p)
         , slow_query_threshold_us(slow_query_threshold)
         , slow_query_ttl_sec(slow_query_ttl)
         , parent_id(std::move(p_id))
+        , classic_tracing(true)
+        , opentelemetry_tracing(otel_tracing)
     {
         state_props.set_if<trace_state_props::write_on_close>(write_on_close);
     }
+
+    explicit trace_info()   // TODO default ctor is not used anywhere, could it be shadowed tho?
```

Comment:
Not sure how to make this better. 
Creating new structure would require passing it through many functions.
Using this one for OpenTelemetry only makes it unsafe (especially because all the attributes are public).

Reply:
Notice the ctor issue.